mlrun/datastore/azure_blob.py

```diff
@@ -74,6 +74,9 @@ def get_storage_options(self):
             or self._get_secret_or_env("AZURE_STORAGE_SAS_TOKEN"),
             credential=self._get_secret_or_env("credential"),
         )
+        # Filter out None values
+        res = {k: v for k, v in res.items() if v}
```

Comment:
This will also filter out empty strings, empty data structures, `False`, and 0 numericals. It looks like only empty strings apply here. Did you mean to filter these out?

Either way, it seems like either the condition or the comment should be updated.

Reply:
Oh, this python...
Anyway - looks like there is a broader issue with the get_storage_options()